platform/backend/src/routes/health.ts

```diff
@@ -0,0 +1,173 @@
+import type { FastifyPluginAsync } from "fastify";
+import config from "../config";
+import db from "../database";
+import { systemHealthStatus } from "../metrics";
+
+const healthRoutes: FastifyPluginAsync = async (fastify) => {
+  fastify.get("/health", async () => {
+    const healthCheck = {
+      status: "ok",
+      timestamp: new Date().toISOString(),
+      uptime: process.uptime(),
+      version: config.api.version,
+      name: config.api.name,
+    };
+
+    systemHealthStatus.set({ component: "system" }, 1);
+
+    return healthCheck;
+  });
+
+  fastify.get("/health/detailed", async () => {
+    const checks: Record<string, unknown> = {
+      system: {
+        status: "ok",
+        timestamp: new Date().toISOString(),
+        uptime: process.uptime(),
+        version: config.api.version,
+        name: config.api.name,
+      },
+      database: { status: "checking" },
+    };
+
+    let overallStatus = "ok";
+
+    try {
+      await db.execute("SELECT 1");
+      checks.database = {
+        status: "ok",
+        message: "Database connection successful",
+      };
+      systemHealthStatus.set({ component: "database" }, 1);
+    } catch (error) {
+      checks.database = {
+        status: "error",
+        message: "Database connection failed",
+        error: error instanceof Error ? error.message : "Unknown error",
+      };
+      overallStatus = "error";
+      systemHealthStatus.set({ component: "database" }, 0);
+    }
+
+    const providerChecks = await checkLLMProviders();
+    checks.providers = providerChecks;
+
+    if (
+      providerChecks.some(
+        (p: { status: string; provider: string }) => p.status === "error",
+      )
+    ) {
+      overallStatus = "degraded";
+    }
+
+    const healthResponse = {
+      status: overallStatus,
+      timestamp: new Date().toISOString(),
+      checks,
+    };
+
+    systemHealthStatus.set(
+      { component: "system" },
+      overallStatus === "ok" ? 1 : 0,
+    );
+
+    return healthResponse;
+  });
+
+  fastify.get("/ready", async () => {
+    try {
+      await db.execute("SELECT 1");
+
+      return {
+        status: "ready",
+        timestamp: new Date().toISOString(),
+      };
+    } catch (_error) {
+      return {
+        status: "not ready",
+        timestamp: new Date().toISOString(),
+        error: "Database not accessible",
+      };
+    }
+  });
+
+  fastify.get("/live", async () => {
+    return {
+      status: "alive",
+      timestamp: new Date().toISOString(),
+      uptime: process.uptime(),
+    };
+  });
```

Comment:
what's the thought behind these three routes, what are they intended for? I don't think we should add routes just to add routes